### PR TITLE
Implemented the advanced search header for the home page

### PR DESCRIPTION
**Pull request by Julyuk**

### CODE REVIEW

File: src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    protected WebElement searchIcon;
+    protected WebElement advancedSearchIcon;
+    protected AdvancedSearchTooltip advancedSearchTooltip;
+    public AdvancedSearchHeaderComponent(WebDriver driver, WebElement rootElement) {
+        super(driver, rootElement);
+    }
+
+    public WebElement getAdvancedSearchTextHeading() {
+        if (advancedSearchTextHeading == null) {
+            advancedSearchTextHeading = rootElement.findElement(By.xpath("//div[contains(@class, \"city-name-box\")]/h2"));
```
Comment (s-hrynus) [inline src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java]:
```suggestion
            advancedSearchTextHeading = rootElement.findElement(By.xpath("//h2[@class='city-name']"));
```
You can use a straight path

File: src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    public WebElement getSelectionSearchInputField() {
+        if (selectionSearchInputField == null) {
+            selectionSearchInputField = rootElement.findElement(By.xpath("//div[contains(@class, \"search\")]//input[@type=\"search\"]"));
+        }
+        return selectionSearchInputField;
+    }
+    public WebElement getSelectionSearchInputFieldPlaceholder() {
+        if (selectionSearchInputFieldPlaceholder == null) {
+            selectionSearchInputFieldPlaceholder = rootElement.findElement(By.xpath("//div[contains(@class, \"search\")]//span[contains(@class,\"ant-select-selection-placeholder\")]"));
```
Comment (s-hrynus) [inline src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java]:
```suggestion
            selectionSearchInputFieldPlaceholder = rootElement.findElement(By.xpath("//span[@class='ant-select-selection-placeholder']"));
```

Same

**Review by OksankaV — APPROVED** [review]

**Review by s-hrynus — APPROVED** [review]
Great job 👍

File: src/main/java/com/academy/ui/pages/BasePage.java [hunk 3]
```diff
@@ -20,6 +23,14 @@ public HeaderComponent getHeader() {
...
         return header;
     }
+    public AdvancedSearchHeaderComponent getAdvancedSearchHeaderComponent() {
+        if (advancedSearchHeaderComponent == null) {
+            WebElement node = driver.findElement(By.xpath("//div[contains(@class, \"lower-header-box\")]"));
+            advancedSearchHeaderComponent = new AdvancedSearchHeaderComponent(driver, node);
+        }
+        return advancedSearchHeaderComponent;
+    }
+
```
Comment (lhalam) [inline src/main/java/com/academy/ui/pages/BasePage.java]: create page BasePageWithAdvancedSearch
move impl to BasePageWithAdvancedSearch

**Pull request merged**